tests/new_return_types/test_parameter_shift_new.py

```diff
@@ -0,0 +1,1115 @@
+    def test_pauli_rotation_gradient(self, mocker, G, theta, shift, tol):
+        """Tests that the automatic gradients of Pauli rotations are correct."""
+
+        qml.enable_return()
+        spy = mocker.spy(qml.gradients.parameter_shift, "_get_operation_recipe")
+        dev = qml.device("default.qubit", wires=1)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QubitStateVector(np.array([1.0, -1.0], requires_grad=False) / np.sqrt(2), wires=0)
+            G(theta, wires=[0])
+            qml.expval(qml.PauliZ(0))
+
+        tape.trainable_params = {1}
+
+        tapes, fn = qml.gradients.param_shift(tape, shifts=[(shift,)])
+        assert len(tapes) == 2
+
+        autograd_val = fn(dev.batch_execute_new(tapes))
+
+        tape_fwd, tape_bwd = tape.copy(copy_operations=True), tape.copy(copy_operations=True)
+        tape_fwd.set_parameters([theta + np.pi / 2])
+        tape_bwd.set_parameters([theta - np.pi / 2])
+
+        manualgrad_val = np.subtract(*dev.batch_execute_new([tape_fwd, tape_bwd])) / 2
+        assert np.allclose(autograd_val, manualgrad_val, atol=tol, rtol=0)
+        assert isinstance(autograd_val, tuple)
+
+        num_params = len(tape.trainable_params)
+        assert len(autograd_val) == num_params
+
+        assert spy.call_args[1]["shifts"] == (shift,)
+
+        # compare to finite differences
+        tapes, fn = qml.gradients.finite_diff(tape)
+        numeric_val = fn(dev.batch_execute_new(tapes))
+        assert np.allclose(autograd_val, numeric_val, atol=tol, rtol=0)
+        qml.disable_return()
+
+    @pytest.mark.parametrize("theta", np.linspace(-2 * np.pi, 2 * np.pi, 7))
+    @pytest.mark.parametrize("shift", [np.pi / 2, 0.3, np.sqrt(2)])
+    def test_Rot_gradient(self, mocker, theta, shift, tol):
+        """Tests that the automatic gradient of an arbitrary Euler-angle-parameterized gate is correct."""
+        qml.enable_return()
+        spy = mocker.spy(qml.gradients.parameter_shift, "_get_operation_recipe")
+        dev = qml.device("default.qubit", wires=1)
+        params = np.array([theta, theta**3, np.sqrt(2) * theta])
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QubitStateVector(np.array([1.0, -1.0], requires_grad=False) / np.sqrt(2), wires=0)
+            qml.Rot(*params, wires=[0])
+            qml.expval(qml.PauliZ(0))
+
+        tape.trainable_params = {1, 2, 3}
+
+        tapes, fn = qml.gradients.param_shift(tape, shifts=[(shift,)] * 3)
+        num_params = len(tape.trainable_params)
+        assert len(tapes) == 2 * num_params
+
+        autograd_val = fn(dev.batch_execute_new(tapes))
+        assert isinstance(autograd_val, tuple)
+        assert len(autograd_val) == num_params
+        manualgrad_val = np.zeros((1, num_params))
+
+        manualgrad_val = []
+        for idx in list(np.ndindex(*params.shape)):
+            s = np.zeros_like(params)
+            s[idx] += np.pi / 2
```

Comment:
Why is this shift fixed and not the same shift that is passed to the test?

Reply:
Good question. This test was copied, I'll stage this comment for now and return later.